drivers/mipi_dbi/mipi_dbi_smartbond.c

```diff
@@ -0,0 +1,610 @@
+	da1469x_lcdc_te_set_status(true, DT_INST_PROP_OR(0, te_polarity, false));
+	/*
+	 * Wait for the TE signal to be asserted so display's refresh status can be synchronized with
+	 * the current frame update.
+	 */
+	k_sem_take(&data->sync_sem, K_FOREVER);
+#endif
+
+	LCDC->LCDC_INTERRUPT_REG |= LCDC_LCDC_INTERRUPT_REG_LCDC_VSYNC_IRQ_EN_Msk;
+
+	/* Setting this bit will enable the host to start outputing pixel data */
+	LCDC->LCDC_MODE_REG |= LCDC_LCDC_MODE_REG_LCDC_SFRAME_UPD_Msk;
+
+	/* Wait for frame update to complete */
+	k_sem_take(&data->sync_sem, K_FOREVER);
+
+	if (data->underflow_flag) {
+		LOG_WRN("Underflow took place");
+		data->underflow_flag = false;
+	}
+}
+
+#if MIPI_DBI_SMARTBOND_IS_RESET_REQUIRED
+static int mipi_dbi_smnartbond_reset(const struct device *dev, uint32_t delay)
+{
+	const struct mipi_dbi_smartbond_config *config = dev->config;
+	int ret;
+
+	if (config->reset.port == NULL) {
+		LOG_WRN("Reset signal not present");
+		return -ENOTSUP;
+	}
+
+	ret = gpio_pin_set_dt(&config->reset, 1);
+	if (ret < 0) {
+		LOG_ERR("Cannot drive reset signal");
+		return ret;
+	}
+	k_msleep(delay);
+
+	return gpio_pin_set_dt(&config->reset, 0);
+}
+#endif
+
+/* Dispplay pixel to output color format translation */
+static inline uint8_t lcdc_smartbond_pixel_to_ocm(enum display_pixel_format pixfmt)
+{
+	switch (pixfmt) {
+	case PIXEL_FORMAT_RGB_565:
+		return (uint8_t)LCDC_SMARTBOND_OCM_RGB565;
+	case PIXEL_FORMAT_RGB_888:
+		return (uint8_t)LCDC_SMARTBOND_OCM_RGB888;
+	case PIXEL_FORMAT_MONO10:
+		return (uint8_t)LCDC_SMARTBOND_L0_L1;
+	default:
+		LOG_ERR("Unsupported pixel format");
+		return 0;
+	};
+}
+
+static inline uint8_t lcdc_smartbond_line_mode_translation(uint8_t mode)
+{
+	switch (mode) {
+	case MIPI_DBI_MODE_SPI_3WIRE:
+		return (uint8_t)LCDC_SMARTBOND_MODE_SPI3;
+	case MIPI_DBI_MODE_SPI_4WIRE:
+		return (uint8_t)LCDC_SMARTBOND_MODE_SPI4;
+	default:
+		LOG_ERR("Unsupported SPI mode");
+		return 0;
+	}
+}
+
+static inline uint8_t lcdc_smartbond_pixel_to_lcm(enum display_pixel_format pixfmt)
+{
+		switch (pixfmt) {
+	case PIXEL_FORMAT_RGB_565:
+		return (uint8_t)LCDC_SMARTBOND_L0_RGB565;
+	case PIXEL_FORMAT_ARGB_8888:
+		return (uint8_t)LCDC_SMARTBOND_L0_ARGB8888;
+	default:
+		LOG_ERR("Unsupported pixel format");
+		return 0;
+	};
+}
+
+static void lcdc_smartbond_mipi_dbi_translation(const struct mipi_dbi_config *dbi_config,
+			lcdc_smartbond_mipi_dbi_cfg *mipi_dbi_cfg,
+			enum display_pixel_format pixfmt)
+{
+	mipi_dbi_cfg->cpha = dbi_config->config.operation & SPI_MODE_CPHA;
+	mipi_dbi_cfg->cpol = dbi_config->config.operation & SPI_MODE_CPOL;
+	mipi_dbi_cfg->cs_active_high = dbi_config->config.operation & SPI_CS_ACTIVE_HIGH;
+	mipi_dbi_cfg->line_mode = lcdc_smartbond_line_mode_translation(dbi_config->mode);
+	mipi_dbi_cfg->color_mode = lcdc_smartbond_pixel_to_ocm(pixfmt);
+}
+
+#if MIPI_DBI_SMARTBOND_IS_READ_SUPPORTED
+static int mipi_dbi_smartbond_command_read(const struct device *dev,
```

Comment:
Just a question here- do you want to support reads? Your driver, and totally up to you. I don't envision every controller supporting reads though, because this is often not a use case required by users- so I just want to make it clear that if you don't see a need for this support, you are welcome to simply not implement it, since the hardware doesn't enable support.

Reply:
Indeed, reading pixel data should not be needed for most use cases. This is why I am safeguarding the read API so it's up to users to exclude it from building or not. Although read operations are not natively supported by our LCDC IP, they can be performed via the standard SPI bus.